src/Dapr.Actors.Generators/Extensions/IEnumerableExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Dapr.Actors.Generators.Extensions
+{
+    internal static class IEnumerableExtensions
+    {
+        /// <summary>
+        /// Returns the index of the first item in the sequence that satisfies the predicate. If no item satisfies the predicate, -1 is returned.
+        /// </summary>
+        /// <typeparam name="T">The type of objects in the <see cref="IEnumerable{T}"/>.</typeparam>
+        /// <param name="source"><see cref="IEnumerable{T}"/> in which to search.</param>
+        /// <param name="predicate">Function performed to check whether an item satisfies the condition.</param>
+        /// <returns>Return the zero-based index of the first occurrence of an element that satisfies the condition, if found; otherwise, -1.</returns>
+        internal static int IndexOf<T>(this IEnumerable<T> source, Func<T, bool> predicate)
+        {
+            if (predicate is null)
+            {
+                throw new ArgumentNullException(nameof(predicate));
+            }
+
+            int index = 0;
+
+            foreach (var item in source)
```

Comment:
Instead of a foreach with a separate variable tracking the index, why not just use a for and repurpose the incrementing value?

Reply:
To avoid enumerating the source parameter twice (one for the Count() and one to iterate source elements), I do not use the `Count()` function that I need to write the `for(int i=0; i < source.Count(); i++)`